src/bin/server/rpc.rs

```diff
@@ -1,19 +1,96 @@
+use std::net::SocketAddr;
+
+use drop::crypto::key::exchange::{self, Exchanger};
+use drop::crypto::sign;
+use drop::net::{ConnectorExt, TcpConnector, TcpListener};
+use drop::system::manager::{Handle, SystemManager};
+use drop::system::sampler::AllSampler;
+use drop::system::sender::NetworkSender;
+use drop::system::System;
+use murmur::MurmurConfig;
+use sieve::{self, Sieve, SieveConfig, SieveMessage};
+
 use tonic::Response;
 
 tonic::include_proto!("at2");
 
 use at2_server::At2;
 pub use at2_server::At2Server;
 
-#[derive(Default)]
-pub struct Service {}
+type M = u64;
+
+pub struct Service {
+    handle: sieve::SieveHandle<M, NetworkSender<SieveMessage<M>>, sieve::Fixed>,
+}
+
+use super::config;
+
+impl Service {
+    pub async fn new(
+        listener_addr: SocketAddr,
+        network_keypair: exchange::KeyPair,
+        sign_keypair: sign::KeyPair,
+        network: Vec<config::Node>,
+    ) -> Self {
+        let exchanger = Exchanger::new(network_keypair);
+
+        let listener = TcpListener::new(listener_addr, exchanger.clone())
+            .await
+            .expect("failed to listen for incoming connections");
```

Comment:
Maybe use [human-panic](https://crates.io/crates/human-panic) for cleaner error messages?